src/hotspot/share/runtime/flags/jvmFlag.cpp

```diff
@@ -701,26 +702,21 @@ void JVMFlag::printFlags(outputStream* out, bool withComments, bool printRanges,
     out->print_cr("[Global flags ranges]");
   }
 
-  // Sort
-  JVMFlag** array = NEW_C_HEAP_ARRAY_RETURN_NULL(JVMFlag*, length, mtArguments);
-  if (array != nullptr) {
-    for (size_t i = 0; i < length; i++) {
-      array[i] = &flagTable[i];
-    }
-    qsort(array, length, sizeof(JVMFlag*), compare_flags);
-
+  // Print the flag with best sort value, then mark it.
+  for (size_t j = 0; j < length; j++) {
+    JVMFlag* best_flag = nullptr;
+    size_t best_i = 0;
     for (size_t i = 0; i < length; i++) {
-      if (array[i]->is_unlocked() && !(skipDefaults && array[i]->is_default())) {
-        array[i]->print_on(out, withComments, printRanges);
+      if (!iteratorMarkers.at(i) && flagTable[i].is_unlocked() && !(skipDefaults && flagTable[i].is_default())) {
```

Comment:
Why does `!(skipDefaults && flagTable[i].is_default())` compute `flagTable[i].is_default()` if `skipDefaults = true`?

Reply:
Because if they are both true we negate and skip. ???

Reply:
I missed that this is pre-existent. Now that I read it, I get it, it really didn't click yesterday. "Check if it's default, and if we should skip defaults then skip it."

Reply:
It would benefit from clearer coding.

E.g. 
```
const bool skip = ...
const bool visited = ...
if (!visited && !skip) ...
```